src/hotspot/share/cds/metaspaceShared.cpp

```diff
@@ -669,6 +669,13 @@ void MetaspaceShared::preload_and_dump(TRAPS) {
       MetaspaceShared::writing_error("Unexpected exception, use -Xlog:cds,exceptions=trace for detail");
     }
   }
+
+  if (AOTMode != nullptr && strcmp(AOTMode, "create") == 0) {
```

Comment:
Would it be better to wrap this test in a CDSConfig::is_ method; which in turn can test the AOTMode FLAG directly or test (CDSConfig::is_dumping_static_archive() && !CDSConfig::is_old_cds_flags_used()), the point being that the meaning of the test is captured in the method name, but the test itself can change over time as needed

Reply:
I added `CDSConfig::is_old_cds_flags_used()` and edited the comment. There's no need to check for `CDSConfig::is_dumping_static_archive()` as this function is called only when dumping the static archive.

(The "if" check may need further tweaking when merging into the Leyden repo, which has several modes of "static" archive dumping).